blocks/carousel/carousel.js

```diff
@@ -0,0 +1,47 @@
+import { button, span } from '../../scripts/dom-helpers.js';
+import { decorateIcons, loadCSS } from '../../scripts/lib-franklin.js';
+import { decorateAPICards } from '../cards/cards.js';
+
+async function decorateAPICarousel(block) {
+  const apiEndpoint = block.querySelector('a').href;
+  block.innerHTML = '';
+  const apiResponse = await fetch(apiEndpoint);
+
+  if (!apiResponse.ok) {
+    return;
+  }
+
+  const apiInfo = JSON.parse(await apiResponse.text());
+  block.append(decorateAPICards(apiInfo.offers, true));
+}
+
+function navLeft(block) {
+  const ul = block.querySelector('ul');
+  ul.scrollLeft -= 600;
+}
+
+function navRight(block) {
+  const ul = block.querySelector('ul');
+  ul.scrollLeft += 600;
+}
+
+export default async function decorate(block) {
+  const cardsCSSPromise = loadCSS('/blocks/cards/cards.css');
+  if (block.children.length === 1 && block.querySelectorAll('a').length === 1) {
+    await decorateAPICarousel(block);
+  }
+
+  if (window.matchMedia('(min-width: 769px)').matches) {
```

Comment:
I don't fully understand. Why is the carousel added just if the screen is larger than `769px`?

Reply:
not the carousel, just the right and left arrow buttons.

Reply:
I got it now, thanks! but still I would personally simply hide those buttons on mobile with something like:
```
@media (max-width: 767px) {
  .carousel .carousel-nav {
    display: none;
 }
}
```
instead of not rendering them, because if for a user opens a small browser window on desktop and then resizes it to full screen, they not have the buttons and would need to refresh. In the end it is not a very likely scenario, so I would not be against your implementation, I just wanted to share the approach I would take.